### ISSUE TITLE
Overriding `export *` causes `TypeError`

### ISSUE THREAD

**Issue by klaussner** [rollup/rollup#3137]
- Rollup Version: 1.21.4
- Operating System (or Browser): macOS 10.14.6
- Node Version: 12.2

### How Do We Reproduce?

[REPL link](https://rollupjs.org/repl/?version=1.21.4&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmV4cG9ydCUyMColMjBmcm9tJTIwJTVDJTIyZnMlNUMlMjIlM0IlNUNuZXhwb3J0JTIwY29uc3QlMjBleGlzdHNTeW5jJTIwJTNEJTIwJTVDJTIyJTNBKSU1QyUyMiUzQiUyMiUyQyUyMmlzRW50cnklMjIlM0F0cnVlJTdEJTVEJTJDJTIyb3B0aW9ucyUyMiUzQSU3QiUyMmZvcm1hdCUyMiUzQSUyMmNqcyUyMiUyQyUyMm5hbWUlMjIlM0ElMjJteUJ1bmRsZSUyMiUyQyUyMmFtZCUyMiUzQSU3QiUyMmlkJTIyJTNBJTIyJTIyJTdEJTJDJTIyZ2xvYmFscyUyMiUzQSU3QiU3RCU3RCUyQyUyMmV4YW1wbGUlMjIlM0FudWxsJTdE).

```js
export * from "fs";
export const existsSync = ":)";
```

### Expected Behavior

The second export overrides `existsSync` of the first export.

### Actual Behavior

A `TypeError` is thrown when the Rollup output is executed in Node:

```
TypeError: Cannot set property existsSync of #<Object> which has only a getter
    at Object.<anonymous> (repl.js:17:20)
    at Module._compile (internal/modules/cjs/loader.js:759:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:770:10)
    at Module.load (internal/modules/cjs/loader.js:628:32)
    at Function.Module._load (internal/modules/cjs/loader.js:555:12)
    at Function.Module.runMain (internal/modules/cjs/loader.js:822:10)
    at internal/main/run_main_module.js:17:11
```

**Comment by lukastaegert** [comment]
I see. So Rollup is basically interfering with its own logic here. I see at least two possible solutions:
- change the reexport namespace copying logic to skip named exports; does not work well for multiple reexports, though
- if a namespace is reexported, subsequent exports should not directly assign to the exports object but also use Object.defineProperty

**Comment by lukastaegert** [comment]
Should also probably check how Babel handles this.

**Comment by klaussner** [comment]
Babel creates an object with all named exports and skips the exports in the copying logic: [Babel REPL link](https://babeljs.io/repl#?babili=false&browsers=&build=&builtIns=false&spec=false&loose=false&code_lz=KYDwDg9gTgLgBAKjgMyhAtnARMgzlgbgChRJY4BjCAO13lAEs7cBlAT2orgF5sAuAJSEiQA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=es2015&prettier=false&targets=&version=7.6.2&externalPlugins=). But it looks like Babel doesn't handle overriding exports correctly, either. It works for named exports but if two `export *` statements are used, the code generated by Babel also fails with a `TypeError` if both modules have exports with the same name because the defined properties are not configurable and cannot be redefined: [Babel REPL link (bug)](https://babeljs.io/repl#?babili=false&browsers=&build=&builtIns=false&spec=false&loose=false&code_lz=KYDwDg9gTgLgBAKjgMyhAtnAROiATAVwBtgBaARiwG4AoUSWRFNTHfYsgJmpqA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=es2015&prettier=false&targets=&version=7.6.2&externalPlugins=).

**Comment by lukastaegert** [comment]
Thanks for checking. Not sure what to make of the second scenario, but at least the first is better than what we have at the moment.

As far as I know, the correct behaviour for conflicting namespace reexports is to remove the conflicting bindings. So if the bindings should be non-configurable, we would need to check and omit the bindings of the other export in such a scenario. I.e.

```js
var _module = require("module-1");
var _module2 = require("module-2");

Object.keys(_module).forEach(function (key) {
  if (key === "default" || key === "__esModule" ||
      Object.prototype.hasOwnProperty.call(_module2, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _module[key];
    }
  });
});

Object.keys(_module2).forEach(function (key) {
  if (key === "default" || key === "__esModule" ||
      Object.prototype.hasOwnProperty.call(_module, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _module2[key];
    }
  });
});
```

Combining this with Babel's approach for the first scenario, I guess this should work, or am I overlooking something here?

**Comment by salmanm** [comment]
As a temporary workaround I've done the following and it works.

```js
import * as fs from "fs";
export const existsSync = ":)";

export default { ...fs, existsSync }
```

But would be good to see this one fixed.

**Comment by aleclarson** [comment]
I'm bumping into this. Is there a better workaround than using `export default`?

**Comment by lukastaegert** [comment]
Sorry for dropping the ball on this. Fix is here: #3959
